docker/Makefile

```diff
@@ -0,0 +1,42 @@
+#
+# Simple Makefile for building docker images
+#
+QNSDIR    = ./qns
+DOCKER    = docker
+
+REPO      = cloudflare/quiche
+BASE_TAG  = latest
+QNS_TAG   = qns-latest
```

Comment:
This is a bit confusing. The targets are different so we should just create two images (`quiche` and `quiche-qns` or something like that), rather than use tags I thhink.

Reply:
IIUC with Docker, the Dockerfile defines multiple images as part of a multi-stage process: `quiche-base` and `quiche-dns` are different images that depend on the common `build`. The make targets `base`  and `qns` run Docker for these targets but I think we want to publish them to unique paths? E.g.

```
BASE_REPO=cloudflare/quiche/quiche-base
QNS_REPO=cloudflare/quiche/quiche-qns

$(DOCKER) push $(BASE_REPO):latest
$(DOCKER) push $(QNS_REPO):latest
```

Reply:
Yeah something like that, but should just be called `quiche` instead of `quiche-base` IMO.

Reply:
Ok. I will split into `cloudflare/quiche` for base and `cloudflare/quiche-dns` for qns.